### ISSUE TITLE
Fix opening ShareX personal folder

### ISSUE THREAD

**Issue by Chefkeks** [ShareX/ShareX#1950]
Hi there,

using ShareX (portable version) now for a while and it's running great! I recently moved ShareX to a new location and since I've set my personal folder to ".\ShareX" I wanted to check whether it worked or it still points to the old location. Under Application settings --> Paths --> ShareX personal folder the path looked pretty good (next to the "Open" button). But when I pressed the "Open" button, it opened the ShareX main window instead of the folder, which I guess is a bug. Would be nice if you could fix that little bug ;)

Beside I think it would be great when a user has set the ShareX personal folder and/or the custom screenshots folder to something like ".\foldername" to display both, relative path and full path next to the open button. To make it a bit better visible, here's a little image showing what I mean:
![image](https://cloud.githubusercontent.com/assets/2871833/18983303/42ef17ce-86ec-11e6-8e76-50fdb959991d.png)

Thanks in advance and best regards,
Michael

**Comment by Jaex** [comment]
I don't understand why you wrote there ".\ShareX". Isn't you are using already portable build? Because in portable build it is already default path. Also I wonder why you wrote there ".\ShareX" instead of "ShareX". Having dot there not changes anything in path, is it not?

**Comment by Chefkeks** [comment]
Thanks for response!

So you're saying in portable version "ShareX" as personal folder path is working and the is no need to prepend ` '\` ?
That would be great!

Ok, just tested before replying and yes it is working with just "ShareX". Hadn't know that and didn't expected this since most other programs on earth expect either a full path or a relative one with ".\foldername" 👍 

Anyway would be great when you could fix the open button of the "ShareX personal folder" since it open's the main window instead the folder. Also when the folders are set "ShareX" / "Screenshots" to show the full location too as I suggested with the image.

Hope you like the idea

Cheers,
Michael

**Comment by Jaex** [comment]
I'm telling no need to write "ShareX" too, Portable already uses it by default. So it should be empty path.

**Comment by Chefkeks** [comment]
Ahh okay, now I get it :D 
Thanks again for clarification!

**Comment by Jaex** [comment]
It already shows full path in preview, I don't understand your suggestion.

**Comment by Jaex** [comment]
But I found few bugs in preview and open buttons gonna fix them now.

**Comment by Chefkeks** [comment]
`Also when the folders are set "ShareX" / "Screenshots" to show the full location too as I suggested with the image.`
` It already shows full path in preview, I don't understand your suggestion. `

You're right, it's my bad - but under "Screenshots" folder the preview does not show the full path, just "Screenshots" or whatever name you use.

`But I found few bugs in preview and open buttons, gonna fix them now.`

So instead of continuing with a long discussion here, I'll let you fix things and test again then, when you've released a new version ;)
Just one more thing: When I leave the field for the personal folder empty, it shows "C:\Users\username\Documents\ShareX" which is wrong. It should be "C:\mysoftware\ShareX\ShareX", at least when using the portable version I think.

So, take your time and let me know when you want me to test something, further feedback or anything else to get things fixed.

**Comment by Jaex** [comment]
Open button issue was caused by this pull request: #1677
What happens is instead of opening "ShareX" folder it actually opens "ShareX.exe" even when there is no ".exe" in path.

And screenshot preview issue was caused by custom screenshot path 2: 66cac748f25460f0e795ba81a44f535b3f749f89
It checks if directory exists in custom screenshot path 1 if not exist it uses path 2. Because directory not exists in your tries. It can't update preview.

Both is not my commit, so I contacted them about it.
